helix-term/src/ui/editor.rs

```diff
@@ -487,17 +487,36 @@ impl EditorView {
             surface.set_string(viewport.x + 1, viewport.y, mode, style);
         }
 
+        let show_encoding = doc.encoding().name() != "UTF-8";
         if let Some(path) = doc.relative_path() {
             let path = path.to_string_lossy();
 
             let title = format!("{}{}", path, if doc.is_modified() { "[+]" } else { "" });
+            let title_len = title.len() as u16;
             surface.set_stringn(
                 viewport.x + 6,
                 viewport.y,
                 title,
                 viewport.width.saturating_sub(6) as usize,
                 style,
             );
+            if show_encoding {
+                surface.set_stringn(
+                    viewport.x + title_len + 8,
+                    viewport.y,
+                    doc.encoding().name(),
+                    16_usize,
```

Comment:
Also, rather than showing this after the filename, it should probably be right aligned next to the line number

Reply:
Can leave this out of the initial PR too if you want